sparta/sparta/collection/Collectable.hpp

```diff
@@ -758,7 +758,35 @@ namespace sparta{
             //! \brief Fill it with the new Name Value pairs from
             //  this cycle of Collection.
             void updateLastRecord_(){
-                last_record_values_ = getDataVector();
+                // These fields only need to be set once - they define the format of the collectable
+                // and remain constant for the duration of the simulation
+                if(SPARTA_EXPECT_FALSE(argos_record_.nameVector.empty())) {
+                    argos_record_.nameVector = getNameStrings();
+                    argos_record_.length = argos_record_.nameVector.size();
+                    has_display_id_field_ = argos_record_.nameVector[0] == "DID";
```

Comment:
I would add parentheses around the == expression. Yes, it's redundant, but it aids readability so that one doesn't scan it and think has_display_id_field_ = argos_record_.nameVector[0] at a rapid glance.

Reply:
Fixed.